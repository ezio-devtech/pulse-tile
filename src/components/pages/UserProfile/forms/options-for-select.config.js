const optionsForThemesField = [{
  value: 'green',
  title: 'Green Theme',
},
{
  value: 'purple',
  title: 'Purple Theme',
},
{
  value: 'gray',
  title: 'Gray Theme',
}];

const optionsForGenderField = [{
  value: 'female',
  title: 'Female',
}, {
  value: 'male',
  title: 'Male',
}];

export { optionsForThemesField, optionsForGenderField }
