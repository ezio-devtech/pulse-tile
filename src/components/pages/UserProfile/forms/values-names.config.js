export const valuesNames = {
  APP_TITLE: 'title',
  LOGO_PATH: 'logoPath',
  SELECT_THEME: 'selectThemeField',
  BROWSER_TITLE: 'browserWindowTitle',
};
