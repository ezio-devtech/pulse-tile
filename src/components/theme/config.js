import vaccinations from './plugins/Vaccinations/index';

/**
 * The list of none-core plugins which are included to the current theme
 *
 * @return array
 */
export const themePlugins = [

    vaccinations,

];

/**
 * The list of clients URLs for none-core plugins
 *
 * @return array
 */
export const themeClientUrls = {

    VACCINATIONS: 'vaccinations',

};

/**
 * The list of pages for none-core plugins
 *
 * @return array
 */
export const themePluginsPages = {

    'vaccinations': {
      breadcrumbs: [{
        title: 'Vaccinations',
        state: '/vaccinations',
      }],
    },

};

/**
 *
 */
export const themePatientSummaryConfig = [
    {
        key: 'vaccinations',
        title: 'Vaccinations',
        titleCheckboxes: 'Vaccinations',
        state: 'vaccinations',
        nameCheckboxes: 'vaccinations',
        imgPreview: '/images/patients-summary/vaccinations.jpg',
        isDefaultSelected: true,
    },
];
