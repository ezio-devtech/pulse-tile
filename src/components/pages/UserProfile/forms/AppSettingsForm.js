import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import FileInput from '../../../form-fields/FileInput';
import { optionsForThemesField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';

@reduxForm({ form: 'appSettingsFormSelector' })
export default class AppSettingsForm extends PureComponent {
  render() {
    return (
      <div className="panel-body-inner">
        <form name="appSettingsForm" className="form">
          <div className="form-group-wrapper">
            <Row>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11}>
                    <Field
                      label="Application Title"
                      name="applicationTitle"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="Application Logo File"
                      name="logoPath"
                      component={FileInput}
                      id="logoPath"
                    />
                    <Field
                      label="Application Themes"
                      name="green"
                      component={Select}
                      options={optionsForThemesField}
                    />
                    <Field
                      label="Browser Window title"
                      name="browserWindowTitle"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
            </Row>
          </div>
        </form>
      </div>)
  }
}
