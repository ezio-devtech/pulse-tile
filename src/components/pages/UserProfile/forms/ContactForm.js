import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import { optionsForCountryField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validateContactForm } from './validation';

@reduxForm({
  form: 'contactFormSelector',
  validate: validateContactForm,
})
export default class ContactForm extends PureComponent {
  render() {
    return (
      <div className="panel-body-inner">
        <form name="personalForm" className="form">
          <div className="form-group-wrapper">
            <Row>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11}>
                    <Field
                      label="Address"
                      name="address"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="City"
                      name="city"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="State"
                      name="state"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="Postal Code"
                      name="postalCode"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="Country"
                      name="country"
                      placeholder=""
                      component={Select}
                      options={optionsForCountryField}
                    />
                  </Col>
                </Row>
              </Col>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11} mdOffset={1}>
                    <Field
                      label="Phone Number"
                      name="phone"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="Email"
                      name="email"
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
            </Row>
          </div>
        </form>
      </div>)
  }
}
