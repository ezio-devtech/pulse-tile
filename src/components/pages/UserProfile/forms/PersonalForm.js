import React, { PureComponent } from 'react';
import { Field, reduxForm } from 'redux-form'
import { Row, Col } from 'react-bootstrap';

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import DateInput from '../../../form-fields/DateInput';
import { optionsForGenderField } from './options-for-select.config';
import Select from '../../../form-fields/SelectFormGroup';
import { validatePersonalForm } from './validation';

@reduxForm({
  form: 'personalFormSelector',
  validate: validatePersonalForm,
})
export default class PersonalForm extends PureComponent {
  render() {
    return (
      <div className="panel-body-inner">
        <form name="personalForm" className="form">
          <div className="form-group-wrapper">
            <Row>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11}>
                    <Field
                      label="First Name"
                      name="firstname"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="Last Name"
                      name="lastname"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                    <Field
                      label="NHS No"
                      name="nhs"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
              <Col xs={12} md={6}>
                <Row>
                  <Col md={11} mdOffset={1}>
                    <Field
                      label="Date of Birth"
                      name="birthday"
                      type="text"
                      placeholder=""
                      component={DateInput}
                    />
                    <Field
                      label="Gender"
                      name="gender"
                      placeholder=""
                      component={Select}
                      options={optionsForGenderField}
                    />
                    <Field
                      label="Doctor"
                      name="doctor"
                      type="text"
                      placeholder=""
                      component={ValidatedInput}
                    />
                  </Col>
                </Row>
              </Col>
            </Row>
          </div>
        </form>
      </div>)
  }
}
