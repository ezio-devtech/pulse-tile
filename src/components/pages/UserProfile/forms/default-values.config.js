import { valuesSettingsForm, valuesPersonalForm, valuesContactForm } from './values-names.config';

const defaultPersonalFormValues = {
  [valuesPersonalForm.FIRST_NAME]: 'Bob',
  [valuesPersonalForm.LAST_NAME]: 'Smith',
  [valuesPersonalForm.SELECT_GENDER]: 'female',
  [valuesPersonalForm.DOCTOR]: 'Dr Emma Huston',
};

const defaultContactFormValues = {
  [valuesContactForm.ADDRESS]: '6801 Tellus Street',
  [valuesContactForm.CITY]: 'Westmorland',
  [valuesContactForm.STATE]: 'Westmorland',
  [valuesContactForm.POSTAL_CODE]: 'Box 306',
  [valuesContactForm.PHONE]: '07624 647524',
  [valuesContactForm.EMAIL]: 'bob.smith@gmail.com',
};

const defaultAppSettingsFormValues = {
  [valuesSettingsForm.APP_TITLE]: 'should be fetched from server',
  [valuesSettingsForm.LOGO_PATH]: 'should be fetched from server',
  [valuesSettingsForm.SELECT_THEME]: 'themeDefault',
  [valuesSettingsForm.BROWSER_TITLE]: 'should be fetched from server',
};

export { defaultPersonalFormValues, defaultContactFormValues, defaultAppSettingsFormValues }
