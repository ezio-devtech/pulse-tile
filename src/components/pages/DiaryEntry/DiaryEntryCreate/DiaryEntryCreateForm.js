import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import { Field, reduxForm, change, formValueSelector } from 'redux-form'

import ValidatedInput from '../../../form-fields/ValidatedInputFormGroup';
import ValidatedTextareaFormGroup from '../../../form-fields/ValidatedTextareaFormGroup';
import SelectFormGroup from '../../../form-fields/SelectFormGroup';
import DateInput from '../../../form-fields/DateInput';
import { validateForm } from '../forms.validation';
import { valuesNames, valuesLabels, noteTypeOptions, formSelectorNames } from '../forms.config';
import { defaultFormValues } from './default-values.config';
import TextareaWithButton from '../../../form-fields/TextareaWithButton';

@reduxForm({
  form: formSelectorNames.DIARY_ENTRIES_CREATE,
  validate: validateForm,
})
class DiaryEntryCreateForm extends PureComponent {
  constructor(props) {
    super(props);
    this.state = {
      recognitionStarted: false,
    }

    this.hasSpeechRecognition = false;
    this.recognition = null;
    try {
      const SpeechRecognition = window.SpeechRecognition || window.webkitSpeechRecognition;
      this.recognition = new SpeechRecognition();
      this.recognition.lang = 'en-US';
      this.recognition.continuous = true;
      this.hasSpeechRecognition = true;
    } catch (error) {
      console.error(error);
    }

    if (this.hasSpeechRecognition) {
      this.recognition.onresult = (event) => {
        const current = event.resultIndex;
        const transcript = event.results[current][0].transcript;
        const currentValue = this.props.notes ? `${this.props.notes} ` : '';
        this.props.dispatch(change(formSelectorNames.DIARY_ENTRIES_CREATE, valuesNames.NOTE, currentValue + transcript));
      }
    }

    this.startSpeach = this.startSpeach.bind(this);
    this.stopSpeach = this.stopSpeach.bind(this);
  }

  componentDidMount() {
    this.props.initialize(defaultFormValues);
  }

  startSpeach(event) {
    event.preventDefault();
    if (this.hasSpeechRecognition && !this.state.recognitionStarted) {
      this.setState({ recognitionStarted: true }, () => this.recognition.start());
    }
  }

  stopSpeach(event) {
    event.preventDefault();
    if (this.hasSpeechRecognition && this.state.recognitionStarted) {
      this.setState({ recognitionStarted: false }, () => this.recognition.stop());
    }
  }

  render() {
    const { isSubmit } = this.props;
    const date = new Date();
    const dateCreated = date.getTime();

    return (
      <div className="panel-body-inner">
        <form name="diaryEntriesCreateForm" className="form">
          <div className="form-group-wrapper">
            <div className="row-expand">
              <div className="col-expand-left">
                <Field
                  label={valuesLabels.TYPE}
                  name={valuesNames.TYPE}
                  id={valuesNames.TYPE}
                  options={noteTypeOptions}
                  component={SelectFormGroup}
                  props={{ isSubmit }}
                />
              </div>
            </div>
            <div className="row-expand">
              <div className="col-expand-left">
                <TextareaWithButton
                  button={
                    !this.state.recognitionStarted ? (
                      <button className="btn btn-success btn-inverse btn-square" onClick={this.startSpeach}><i
                        className="fa fa-microphone"></i></button>
                    ) : (
                      <button className="btn btn-success btn-inverse btn-square active" onClick={this.stopSpeach}><i
                        className="fa fa-microphone"></i></button>
                    )
                  }
                  fieldProps={
                    {
                      label: valuesLabels.NOTE,
                      name: valuesNames.NOTE,
                      id: valuesNames.NOTE,
                      component: ValidatedTextareaFormGroup,
                      props: { isSubmit },
                    }
                  }
                />
              </div>
            </div>
            <div className="row-expand">
              <div className="col-expand-left">
                <Field
                  label={valuesLabels.AUTHOR}
                  name={valuesNames.AUTHOR}
                  id={valuesNames.AUTHOR}
                  component={ValidatedInput}
                  props={{ disabled: true }}
                />
              </div>
              <div className="col-expand-right">
                <Field
                  label={valuesLabels.DATE}
                  name={valuesNames.DATE}
                  id={valuesNames.DATE}
                  component={DateInput}
                  props={{ disabled: true, value: dateCreated, format: 'DD-MMM-YYYY' }}
                />
              </div>
            </div>
          </div>
        </form>
      </div>)
  }
}

export default connect((state) => {
  const selector = formValueSelector(formSelectorNames.DIARY_ENTRIES_CREATE);
  return {
    notes: selector(state, valuesNames.NOTE),
  };
})(DiaryEntryCreateForm)
