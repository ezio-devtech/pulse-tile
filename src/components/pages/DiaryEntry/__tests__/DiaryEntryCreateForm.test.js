import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import configureStore from 'redux-mock-store';
import { SpeechRecognitionMock } from 'speech-recognition-mock';

import DiaryEntryCreateForm from '../DiaryEntryCreate/DiaryEntryCreateForm';
import { valuesNames, valuesLabels } from '../forms.config';

Enzyme.configure({adapter: new Adapter()});

const mockStore = configureStore();
const store = mockStore({});
const DATE_TO_USE = new Date('2017');
const DATE_TO_USE_TIME = DATE_TO_USE.getTime();
global.Date = jest.fn(() => DATE_TO_USE);

const FORM_NAME = 'diaryEntriesCreateForm';
const DATE_FORMAT = 'DD-MMM-YYYY';

const testProps = {
  notes: '',
  isSubmit: false,
};

describe('Component <DiaryEntryCreateForm />', () => {
  it('should renders with props correctly', () => {
    const component = shallow(
      <DiaryEntryCreateForm
        store={store}
        isSubmit={testProps.isSubmit}
      />).dive().dive().dive().dive();

    expect(component.find('div.row-expand')).toHaveLength(3);

    expect(component.find('Field')).toHaveLength(3);
    expect(component.find('form')).toHaveLength(1);
    expect(component.find('form').prop('name')).toEqual(FORM_NAME);

    expect(component.find('Field').at(0).props().name).toEqual(valuesNames.TYPE);
    expect(component.find('Field').at(0).props().id).toEqual(valuesNames.TYPE);
    expect(component.find('Field').at(0).props().label).toEqual(valuesLabels.TYPE);
    expect(component.find('Field').at(0).props().props.isSubmit).toEqual(false);

    expect(component.find('Field').at(1).props().name).toEqual(valuesNames.AUTHOR);
    expect(component.find('Field').at(1).props().id).toEqual(valuesNames.AUTHOR);
    expect(component.find('Field').at(1).props().label).toEqual(valuesLabels.AUTHOR);
    expect(component.find('Field').at(1).props().props.disabled).toEqual(true);

    expect(component.find('Field').at(2).props().name).toEqual(valuesNames.DATE);
    expect(component.find('Field').at(2).props().label).toEqual(valuesLabels.DATE);
    expect(component.find('Field').at(2).props().props.disabled).toEqual(true);
    expect(component.find('Field').at(2).props().props.value).toEqual(DATE_TO_USE_TIME);
    expect(component.find('Field').at(2).props().props.format).toEqual(DATE_FORMAT);

    expect(component).toMatchSnapshot();
  });

  it('should renders correctly when form is submitted', () => {
    const component = shallow(
      <DiaryEntryCreateForm
        store={store}
        isSubmit
      />).dive().dive().dive().dive();
    expect(component.find('Field')).toHaveLength(3);
    expect(component.find('form')).toHaveLength(1);
    expect(component.find('form').prop('name')).toEqual(FORM_NAME);

    expect(component.find('Field').at(0).props().props.isSubmit).toEqual(true);
    expect(component.find('TextareaWithButton').at(0).props().fieldProps.props.isSubmit).toEqual(true);

    expect(component).toMatchSnapshot();
  });

  it('should not work when no speech recognition in browser', () => {
    window.SpeechRecognition = null;
    const component = shallow(
      <DiaryEntryCreateForm
        store={store}
        isSubmit
      />).dive().dive().dive().dive();
    const speechInput = component.find('TextareaWithButton');
    const speechButton = speechInput.dive('button');

    expect(speechInput).toHaveLength(1);
    expect(speechButton).toHaveLength(1);

    // will allways be false because no speech recognition in browser during tests
    speechButton.first().simulate('click', {
      preventDefault() {
      }
    });
    expect(component.state().recognitionStarted).toEqual(false);

    component.instance().stopSpeach({
      preventDefault() {
      }
    });
    expect(component.state().recognitionStarted).toEqual(false);

    expect(component).toMatchSnapshot();
  });
  it('should work when speech recognition in browser is active', () => {
    const speechRecognitionEventMock = {
      resultIndex: 0,
      results: [
        [
          {
            transcript: 'test',
          },
        ],
      ],
    };
    window.SpeechRecognition = SpeechRecognitionMock;
    const component = shallow(
      <DiaryEntryCreateForm
        store={store}
        isSubmit
      />).dive().dive().dive().dive();
    const speechInput = component.find('TextareaWithButton');
    const speechButton = speechInput.dive('button');
    const spy = spyOn(store, 'dispatch');

    expect(speechInput).toHaveLength(1);
    expect(speechButton).toHaveLength(1);

    component.instance().startSpeach({
      preventDefault() {
      }
    });
    expect(component.state().recognitionStarted).toEqual(true);

    component.instance().stopSpeach({
      preventDefault() {
      }
    });
    expect(component.state().recognitionStarted).toEqual(false);
    new Promise(()=>{
      component.instance().recognition.onresult(speechRecognitionEventMock)
    }).then(()=>{
      expect(spy).toHaveBeenCalled();
    });
    expect(component).toMatchSnapshot();
  });
});
