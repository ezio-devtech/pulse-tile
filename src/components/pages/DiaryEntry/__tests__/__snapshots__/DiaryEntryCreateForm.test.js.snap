// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <DiaryEntryCreateForm /> should not work when no speech recognition in browser 1`] = `
<div
  className="panel-body-inner"
>
  <form
    className="form"
    name="diaryEntriesCreateForm"
  >
    <div
      className="form-group-wrapper"
    >
      <div
        className="row-expand"
      >
        <div
          className="col-expand-left"
        >
          <Field
            component={[Function]}
            id="noteType"
            label="Type"
            name="noteType"
            options={
              Array [
                Object {
                  "title": "Exam Report",
                  "value": "Exam Report",
                },
                Object {
                  "title": "Note",
                  "value": "Note",
                },
                Object {
                  "title": "Hypothesis",
                  "value": "Hypothesis",
                },
              ]
            }
            props={
              Object {
                "isSubmit": true,
              }
            }
          />
        </div>
      </div>
      <div
        className="row-expand"
      >
        <div
          className="col-expand-left"
        >
          <TextareaWithButton
            button={
              <button
                className="btn btn-success btn-inverse btn-square"
                onClick={[Function]}
              >
                <i
                  className="fa fa-microphone"
                />
              </button>
            }
            fieldProps={
              Object {
                "component": [Function],
                "id": "notes",
                "label": "Note",
                "name": "notes",
                "props": Object {
                  "isSubmit": true,
                },
              }
            }
          />
        </div>
      </div>
      <div
        className="row-expand"
      >
        <div
          className="col-expand-left"
        >
          <Field
            component={[Function]}
            id="author"
            label="Author"
            name="author"
            props={
              Object {
                "disabled": true,
              }
            }
          />
        </div>
        <div
          className="col-expand-right"
        >
          <Field
            component={[Function]}
            id="currentDate"
            label="Date"
            name="currentDate"
            props={
              Object {
                "disabled": true,
                "format": "DD-MMM-YYYY",
                "value": 1483228800000,
              }
            }
          />
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`Component <DiaryEntryCreateForm /> should renders correctly when form is submitted 1`] = `
<div
  className="panel-body-inner"
>
  <form
    className="form"
    name="diaryEntriesCreateForm"
  >
    <div
      className="form-group-wrapper"
    >
      <div
        className="row-expand"
      >
        <div
          className="col-expand-left"
        >
          <Field
            component={[Function]}
            id="noteType"
            label="Type"
            name="noteType"
            options={
              Array [
                Object {
                  "title": "Exam Report",
                  "value": "Exam Report",
                },
                Object {
                  "title": "Note",
                  "value": "Note",
                },
                Object {
                  "title": "Hypothesis",
                  "value": "Hypothesis",
                },
              ]
            }
            props={
              Object {
                "isSubmit": true,
              }
            }
          />
        </div>
      </div>
      <div
        className="row-expand"
      >
        <div
          className="col-expand-left"
        >
          <TextareaWithButton
            button={
              <button
                className="btn btn-success btn-inverse btn-square"
                onClick={[Function]}
              >
                <i
                  className="fa fa-microphone"
                />
              </button>
            }
            fieldProps={
              Object {
                "component": [Function],
                "id": "notes",
                "label": "Note",
                "name": "notes",
                "props": Object {
                  "isSubmit": true,
                },
              }
            }
          />
        </div>
      </div>
      <div
        className="row-expand"
      >
        <div
          className="col-expand-left"
        >
          <Field
            component={[Function]}
            id="author"
            label="Author"
            name="author"
            props={
              Object {
                "disabled": true,
              }
            }
          />
        </div>
        <div
          className="col-expand-right"
        >
          <Field
            component={[Function]}
            id="currentDate"
            label="Date"
            name="currentDate"
            props={
              Object {
                "disabled": true,
                "format": "DD-MMM-YYYY",
                "value": 1483228800000,
              }
            }
          />
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`Component <DiaryEntryCreateForm /> should renders with props correctly 1`] = `
<div
  className="panel-body-inner"
>
  <form
    className="form"
    name="diaryEntriesCreateForm"
  >
    <div
      className="form-group-wrapper"
    >
      <div
        className="row-expand"
      >
        <div
          className="col-expand-left"
        >
          <Field
            component={[Function]}
            id="noteType"
            label="Type"
            name="noteType"
            options={
              Array [
                Object {
                  "title": "Exam Report",
                  "value": "Exam Report",
                },
                Object {
                  "title": "Note",
                  "value": "Note",
                },
                Object {
                  "title": "Hypothesis",
                  "value": "Hypothesis",
                },
              ]
            }
            props={
              Object {
                "isSubmit": false,
              }
            }
          />
        </div>
      </div>
      <div
        className="row-expand"
      >
        <div
          className="col-expand-left"
        >
          <TextareaWithButton
            button={
              <button
                className="btn btn-success btn-inverse btn-square"
                onClick={[Function]}
              >
                <i
                  className="fa fa-microphone"
                />
              </button>
            }
            fieldProps={
              Object {
                "component": [Function],
                "id": "notes",
                "label": "Note",
                "name": "notes",
                "props": Object {
                  "isSubmit": false,
                },
              }
            }
          />
        </div>
      </div>
      <div
        className="row-expand"
      >
        <div
          className="col-expand-left"
        >
          <Field
            component={[Function]}
            id="author"
            label="Author"
            name="author"
            props={
              Object {
                "disabled": true,
              }
            }
          />
        </div>
        <div
          className="col-expand-right"
        >
          <Field
            component={[Function]}
            id="currentDate"
            label="Date"
            name="currentDate"
            props={
              Object {
                "disabled": true,
                "format": "DD-MMM-YYYY",
                "value": 1483228800000,
              }
            }
          />
        </div>
      </div>
    </div>
  </form>
</div>
`;

exports[`Component <DiaryEntryCreateForm /> should work when speech recognition in browser is active 1`] = `
<div
  className="panel-body-inner"
>
  <form
    className="form"
    name="diaryEntriesCreateForm"
  >
    <div
      className="form-group-wrapper"
    >
      <div
        className="row-expand"
      >
        <div
          className="col-expand-left"
        >
          <Field
            component={[Function]}
            id="noteType"
            label="Type"
            name="noteType"
            options={
              Array [
                Object {
                  "title": "Exam Report",
                  "value": "Exam Report",
                },
                Object {
                  "title": "Note",
                  "value": "Note",
                },
                Object {
                  "title": "Hypothesis",
                  "value": "Hypothesis",
                },
              ]
            }
            props={
              Object {
                "isSubmit": true,
              }
            }
          />
        </div>
      </div>
      <div
        className="row-expand"
      >
        <div
          className="col-expand-left"
        >
          <TextareaWithButton
            button={
              <button
                className="btn btn-success btn-inverse btn-square"
                onClick={[Function]}
              >
                <i
                  className="fa fa-microphone"
                />
              </button>
            }
            fieldProps={
              Object {
                "component": [Function],
                "id": "notes",
                "label": "Note",
                "name": "notes",
                "props": Object {
                  "isSubmit": true,
                },
              }
            }
          />
        </div>
      </div>
      <div
        className="row-expand"
      >
        <div
          className="col-expand-left"
        >
          <Field
            component={[Function]}
            id="author"
            label="Author"
            name="author"
            props={
              Object {
                "disabled": true,
              }
            }
          />
        </div>
        <div
          className="col-expand-right"
        >
          <Field
            component={[Function]}
            id="currentDate"
            label="Date"
            name="currentDate"
            props={
              Object {
                "disabled": true,
                "format": "DD-MMM-YYYY",
                "value": 1483228800000,
              }
            }
          />
        </div>
      </div>
    </div>
  </form>
</div>
`;
